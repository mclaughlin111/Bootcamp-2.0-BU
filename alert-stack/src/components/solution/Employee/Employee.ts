export type Employee = {
  id: number;
  firstName: string;
  surName: string;
  gender: "male" | "female";
  company: string;
  email: string;
  phone: string;
  address: string;
  team: "CD&A" | "CL" | "CM&P" | "CS&E";
};
