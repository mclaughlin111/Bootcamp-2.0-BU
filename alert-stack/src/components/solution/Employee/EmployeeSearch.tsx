import { useState } from "react";

type EmployeeSearchProps = {
  onSubmitSearch: (searchText: string) => void;
};

export const EmployeeSearch = ({ onSubmitSearch }: EmployeeSearchProps) => {
  const [searchText, setSearchText] = useState("");

  return (
    <>
      <input
        type="text"
        onChange={(event) => setSearchText(event.target.value)}
        onKeyUp={({ key }) => key === "Enter" && onSubmitSearch(searchText)}
      />
      <button
        onClick={() => {
          onSubmitSearch(searchText);
        }}
      >
        Search
      </button>
    </>
  );
};
