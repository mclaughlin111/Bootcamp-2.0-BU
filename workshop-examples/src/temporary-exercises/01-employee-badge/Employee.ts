export type Employee = {
  details: {
    firstName: string;
    middleName?: string;
    lastName: string;
  };
  imagePath: string;
  jobTitle: string;
  email: string;
  cohort?: 12 | 13;
  team:
    | "Client Servicing and Engagement"
    | "Cash Management and Payments"
    | "Commercial Lending Platform"
    | "Client Data and Analytics";
};
