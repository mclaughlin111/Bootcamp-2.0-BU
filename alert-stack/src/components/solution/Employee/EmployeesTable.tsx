import {
  TableContainer,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
} from "@mui/material";
import { Employee } from "./Employee";

type EmployeesTableProps = {
  employees: Employee[];
};

export const EmployeesTable = ({ employees }: EmployeesTableProps) => {
  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>First Name</TableCell>
            <TableCell>Last Name</TableCell>
            <TableCell>Company</TableCell>
            <TableCell>Email</TableCell>
            <TableCell>Phone</TableCell>
            <TableCell>Address</TableCell>
            <TableCell>Team</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {employees.map((employee) => {
            const {
              id,
              firstName,
              surName,
              company,
              email,
              phone,
              address,
              team,
            } = employee;
            return (
              <TableRow key={id}>
                <TableCell>{firstName}</TableCell>
                <TableCell>{surName}</TableCell>
                <TableCell>{company}</TableCell>
                <TableCell>{email}</TableCell>
                <TableCell>{phone}</TableCell>
                <TableCell>{address}</TableCell>
                <TableCell>{team}</TableCell>
              </TableRow>
            );
          })}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

// import {
//     Paper,
//     Table,
//     TableBody,
//     TableCell,
//     TableContainer,
//     TableHead,
//     TableRow,
//   } from "@mui/material";
//   import { Employee } from "./Employee";

//   export const EmployeesTable = ({ employees }: { employees: Employee[] }) => {
//     return (
//       <TableContainer component={Paper}>
//         <Table sx={{ minWidth: 650 }} aria-label="simple table">
//           <TableHead>
//             <TableRow>
//               <TableCell>First Name</TableCell>
//               <TableCell>Last Name</TableCell>
//               <TableCell>Company</TableCell>
//               <TableCell>Email</TableCell>
//               <TableCell>Phone</TableCell>
//               <TableCell>Address</TableCell>
//               <TableCell>Team</TableCell>
//             </TableRow>
//           </TableHead>
//           <TableBody>
