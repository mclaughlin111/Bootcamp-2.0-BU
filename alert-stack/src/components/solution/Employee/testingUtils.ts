import { Employee } from "./Employee";

export const testEmployees: Employee[] = [
  {
    id: 1,
    firstName: "Carol",
    surName: "Pugh",
    gender: "female",
    company: "PYRAMIS",
    email: "carolpugh@pyramis.com",
    phone: "+1 (894) 469-2330",
    address: "396 Gilmore Court, Edenburg, American Samoa, 4850",
    team: "CL",
  },
  {
    id: 2,
    firstName: "Camacho",
    surName: "Abbott",
    gender: "male",
    company: "NAXDIS",
    email: "camachoabbott@naxdis.com",
    phone: "+1 (839) 473-3027",
    address: "541 Bay Parkway, Hampstead, Massachusetts, 7398",
    team: "CD&A",
  },
  {
    id: 3,
    firstName: "Cecilia",
    surName: "Herring",
    gender: "female",
    company: "GEEKETRON",
    email: "ceciliaherring@geeketron.com",
    phone: "+1 (866) 584-2623",
    address: "625 Royce Place, Ticonderoga, Maryland, 3541",
    team: "CM&P",
  },
];
