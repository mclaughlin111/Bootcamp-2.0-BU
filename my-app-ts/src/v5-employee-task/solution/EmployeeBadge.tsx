import { Employee } from "../Employee";
import styles from "./EmployeeBadge.module.css";

export const EmployeeBadge = (employee: Employee) => {
  const {
    cohort,
    details: { firstName, middleName, lastName },
    email,
    jobTitle,
    team,
    imagePath,
  } = employee;
  return (
    <div className={styles.employeeBadge}>
      <img alt={`${firstName} ${lastName}`} src={`/images/${imagePath}`} />
      <div>
        <h2>
          {firstName} {middleName ?? ""} {lastName}
        </h2>
        <div>Job Title: {jobTitle}</div>
        <div>
          <span>Email:</span>
          <a href={`mailto:${email}`}>{email}</a>
        </div>
        <div>Cohort: {cohort ?? "-"}</div>
        <div>Team: {team}</div>
      </div>
    </div>
  );
};
