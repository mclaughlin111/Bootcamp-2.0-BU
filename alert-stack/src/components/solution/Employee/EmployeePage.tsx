import { useState } from "react";
import { EmployeeSearch } from "./EmployeeSearch";
import { EmployeesTable } from "./EmployeesTable";
import { Employee } from "./Employee";
import { useQuery } from "@tanstack/react-query";
import axios from "axios";

export const EmployeePage = () => {
  const [searchText, setSearchText] = useState("");

  const {
    isLoading,
    isSuccess,
    data: result,
  } = useQuery({
    queryKey: ["employees", searchText],
    queryFn: () => axios.get<Employee[]>(`/employees?search=${searchText}`),
  });

  return (
    <>
      <EmployeeSearch onSubmitSearch={setSearchText} />
      {isLoading ? (
        <div>Is Loading</div>
      ) : (
        isSuccess && <EmployeesTable employees={result.data} />
      )}
    </>
  );
};
