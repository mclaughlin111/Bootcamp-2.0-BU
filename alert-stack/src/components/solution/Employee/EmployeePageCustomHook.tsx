import { useState } from "react";
import { EmployeeSearch } from "./EmployeeSearch";
import { EmployeesTable } from "./EmployeesTable";
import { useEmployees } from "./useEmployees";

export const EmployeePageCustomHook = () => {
  const [searchText, setSearchText] = useState("");

  const { employees, isLoading } = useEmployees(searchText);

  return (
    <>
      <EmployeeSearch onSubmitSearch={setSearchText} />
      {isLoading ? (
        <div>Is Loading</div>
      ) : (
        <EmployeesTable employees={employees} />
      )}
    </>
  );
};
