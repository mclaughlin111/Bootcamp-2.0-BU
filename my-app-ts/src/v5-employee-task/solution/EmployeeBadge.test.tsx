/* eslint-disable testing-library/no-debugging-utils */
import { render } from "@testing-library/react";
import { EmployeeBadge } from "./EmployeeBadge";
import { Employee } from "../Employee";

const testEmployees: Employee[] = [
  {
    details: {
      firstName: "Duncan",
      middleName: "Andrews",
      lastName: "Edwards",
    },
    imagePath: "images/duncan.jpg",
    jobTitle: "Junior Developer",
    email: "duncan.edwards@lloydsbanking.com",
    cohort: 12,
    team: "Client Servicing and Engagement",
  },
  {
    details: {
      firstName: "Joan",
      lastName: "Smith",
    },
    imagePath: "images/joan.jpg",
    jobTitle: "Seniror Developer",
    email: "joan.smith@lloydsbanking.com",
    team: "Commercial Lending Platform",
  },
];

test.each<Employee>(testEmployees)("Renders an employee badge", (employee) => {
  const { container } = render(<EmployeeBadge {...employee} />);

  expect(container).toMatchSnapshot();
});
