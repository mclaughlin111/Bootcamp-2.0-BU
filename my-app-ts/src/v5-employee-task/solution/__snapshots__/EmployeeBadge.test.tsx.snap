// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an employee badge 1`] = `
<div>
  <div
    class="employeeBadge"
  >
    <img
      alt="Duncan Edwards"
      src="/images/images/duncan.jpg"
    />
    <div>
      <h2>
        Duncan
         
        Andrews
         
        Edwards
      </h2>
      <div>
        Job Title2: 
        Junior Developer
      </div>
      <div>
        <a
          href="mailto:duncan.edwards@lloydsbanking.com"
        >
          duncan.edwards@lloydsbanking.com
        </a>
      </div>
      <div>
        Cohort: 
        12
      </div>
      <div>
        Team: 
        Client Servicing and Engagement
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders an employee badge 2`] = `
<div>
  <div
    class="employeeBadge"
  >
    <img
      alt="Joan Smith"
      src="/images/images/joan.jpg"
    />
    <div>
      <h2>
        Joan
         
         
        Smith
      </h2>
      <div>
        Job Title2: 
        Seniror Developer
      </div>
      <div>
        <a
          href="mailto:joan.smith@lloydsbanking.com"
        >
          joan.smith@lloydsbanking.com
        </a>
      </div>
      <div>
        Cohort: 
        -
      </div>
      <div>
        Team: 
        Commercial Lending Platform
      </div>
    </div>
  </div>
</div>
`;
